chain/client/src/client_actor.rs

```diff
@@ -229,56 +230,23 @@ impl Handler<NetworkClientMessages> for ClientActor {
                 }
             }
             NetworkClientMessages::StateRequest(shard_id, hash, need_header, parts, route_back) => {
-                let mut data = vec![];
-                for part_id in parts.ids.iter() {
-                    match self.client.chain.get_state_response_part(
-                        shard_id,
-                        *part_id,
-                        parts.num_parts,
-                        hash,
-                    ) {
-                        Ok(part) => data.push(part),
-                        Err(e) => {
-                            error!(target: "sync", "Cannot build sync part (get_state_response_part): {}", e);
-                            return NetworkClientResponses::NoResponse;
-                        }
-                    }
-                }
-                if need_header {
-                    match self.client.chain.get_state_response_header(shard_id, hash) {
-                        Ok(header) => {
-                            return NetworkClientResponses::StateResponse(
-                                StateResponseInfo {
-                                    shard_id,
-                                    hash,
-                                    shard_state: ShardStateSyncResponse {
-                                        header: Some(header),
-                                        part_ids: parts.ids,
-                                        data,
-                                    },
-                                },
-                                route_back,
-                            );
-                        }
-                        Err(e) => {
-                            error!(target: "sync", "Cannot build sync header (get_state_response_header): {}", e);
-                            return NetworkClientResponses::NoResponse;
-                        }
-                    }
-                } else {
-                    return NetworkClientResponses::StateResponse(
-                        StateResponseInfo {
-                            shard_id,
-                            hash,
-                            shard_state: ShardStateSyncResponse {
-                                header: None,
-                                part_ids: parts.ids,
-                                data,
-                            },
-                        },
-                        route_back,
-                    );
-                }
+                let network_adapter = self.network_adapter.clone();
+                // TODO discuss vulnerability: a node can be flooded by StateRequests
+                actix::spawn(
```

Comment:
It will still compete for resources with client. We need to remove this response to `ViewClient`. I can do that change.

Reply:
How does it help? Which queries we also should move to `ViewClient`?

Reply:
Because ViewClient is run in a separate thread. Actually that will be a much larger refactoring so I'll do it in a separate PR. We can merge this one first.

Reply:
@bowenwang1996 I still cannot understand clearly the benefit of removing the code from ClientActor to ViewClient. As ViewClient runs in separate thread, does it compete for resources with ClientActor? If yes, what is the benefit of removing? If no, how does it implement?

I guess you have understandable arguments and I'm ready to accept them. The question then, does this PR help you to complete the task. I see two options here:
1. The difference is huge, so you will start with staging.
- However, if the fix takes time, we can use this one as baseline.
- Otherwise it's better to close the PR with no merging and assign you to the issue.
2. This PR is a good start to further refactoring.
- We can merge it now and you will be asked to implement limitations from this comment: https://github.com/nearprotocol/nearcore/pull/1916#issuecomment-570663986
- I can complete them and send to the PR.

Please share your thoughts.